### PR TITLE
Create Astro plugin

### PR DESCRIPTION
**Pull request by sr5434**
Same as pull #1633 on Significant-Gravitas/Auto-GPT but as a plugin. It was approved by @nponeccop but I was asked to reimplement it as a plugin. I will include the old pull request message below.

### Background
The current agent cannot access real-time information about how many astronauts are in space.

### Changes

This uses the open notify api to get information about how many astronauts are in space.

### Documentation

I included a docstring for the function that fetches the data.

### Test Plan

I haven’t tested it, as I added my code on the main branch which is currently broken

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes 

I didn’t add tests because I forked master which is broken

### CODE REVIEW

**Review by ntindle — CHANGES_REQUESTED** [review]
Add tests and improve readme with some more context. Take a look at the email plug-in for examples

**Comment by ntindle** [comment]
Really solid start!

**Comment by sr5434** [comment]
Is this better? Should I squash the commits?

**Comment by ntindle** [comment]
This looks way better! I’ll be able to run/test it a bit later today. It’s 8am for me so by around 2am it should be reviewed

**Comment by sr5434** [comment]
Ok, thanks

File: README.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ Some third-party plugins have been created by contributors that are not included
 | Telegram | A smoothly working Telegram bot that gives you all the messages you would normally get through the Terminal. | [Wladastic/Auto-GPT-Telegram-Plugin](https://github.com/Wladastic/Auto-GPT-Telegram-Plugin) |
 | MetaTrader | Connect your MetaTrader Account to Auto-GPT. | [isaiahbjork/Auto-GPT-MetaTrader-Plugin](https://github.com/isaiahbjork/Auto-GPT-MetaTrader-Plugin) |
 | Google Analytics | Connect your Google Analytics Account to Auto-GPT. | [isaiahbjork/Auto-GPT-Google-Analytics-Plugin](https://github.com/isaiahbjork/Auto-GPT-Google-Analytics-Plugin)
+| YouTube   | Various YouTube features including downloading and understanding | [jpetzke/AutoGPT-YouTube](https://github.com/jpetzke/AutoGPT-YouTube)
```
[the remark this answers is not in the document]
Comment (sr5434) [reply README.md:73]: Those are all 3rd party plugins. Mine is a first-party plugin.

[the remark this answers is not in the document]
Comment (ntindle) [reply README.md:73]: i can modify this if you're okay with it to add it

[the remark this answers is not in the document]
Comment (sr5434) [reply README.md:73]: Oh I didn’t see it. I can add it. Should I squash the commits or leave them?

File: src/autogpt_plugins/astro/test_astro_plugin.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from email_plugin import (
+    get_num_astronauts
+)
```
Comment (ntindle) [inline src/autogpt_plugins/astro/test_astro_plugin.py:3]: import from the correct module

File: src/autogpt_plugins/astro/test_astro_plugin.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import unittest
```
Comment (ntindle) [inline src/autogpt_plugins/astro/test_astro_plugin.py:4]: Use pytest

**Review by ntindle — CHANGES_REQUESTED** [review]
Only small test changes left!

File: src/autogpt_plugins/astro/test_astro_plugin.py (hunk 2 above)
Comment (sr5434) [reply src/autogpt_plugins/astro/test_astro_plugin.py:3]: Ok

**Comment by sr5434** [comment]
Is this better?

**Comment by sr5434** [comment]
@ntindle the tests pass when I ran them. Will this be merged?

**Comment by ntindle** [comment]
Will get to this within the next 17 hours

File: README.md [hunk 4]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (ntindle) [inline README.md:1]: Merge artifact

File: README.md [hunk 5]
```diff
@@ -111,3 +114,7 @@ Creating a plugin is a rewarding experience! You can choose between first-party
 ## Get Help
 
 Visit the [discord](https://discord.gg/autogpt) server for more information.
+=======
+# Auto-GPT-Plugin-Template
```
Comment (ntindle) [inline README.md:118]: Merge artifact

**Review by ntindle — CHANGES_REQUESTED** [review]
Remove the merge conflict artifact or I can when I get time to run the tests

**Comment by ntindle** [comment]
I cannot for the life of me get it to execute the command. Try changing the name/description up or provide a prompt that I can use

**Comment by sr5434** [comment]
Ok, sure.

**Comment by sr5434** [comment]
I used the email plugin __init__.py as a template, and forgot to edit the description. It should(hopefully) work now.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team to all open Plugin-PRs

We're in the process of re-architecting:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

Please be patient as we work to clean up the AutoGPT core.

If you already have an existing plug-in that’s affected, please message @ntindle on Discord so we can get your plugin spinning on the new 
architecture.

If you're working on a plugin, please bear in mind that one of our primary goals in re-architecting is to make life MUCH easier for you. 
We've taken aboard feedback from plugin developers. We're on it, so please give us some time!

<3 The AutoGPT Team

**Pull request merged**